arches_querysets/tasks.py

```diff
@@ -0,0 +1,12 @@
+from arches.app.models.resource import Resource
+from celery import shared_task
+
+
+@shared_task
+def index_resource(resource_id):
+    proxy_resource = (
+        Resource.objects.filter(pk=resource_id)
+        # .select_related("graph__publication")
+        .get()
+    )
+    proxy_resource.index()
```

Comment:
maybe we should add a little error trapping here

```suggestion
import logging
from arches.app.models.resource import Resource
from celery import shared_task


@shared_task
def index_resource(resource_id):
    logger = logging.getLogger(__name__)
    try:
        proxy_resource = (
            Resource.objects.get(pk=resource_id)
        )
        proxy_resource.index()
    except Resource.DoesNotExist:
        logger.error(f"Resource with id {resource_id} does not exist.")
    except Exception as e:
        logger.error(f"Error indexing resource {resource_id}: {e}")

```